src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -194,6 +194,21 @@ impl Coordinator {
                 }
                 _ => None,
             };
+
+            // Attempt to reduce the `CHECK` expression, we timeout if this takes too long.
+            if let mz_sql::plan::DataSourceDesc::Webhook {
+                validate_using: Some(validate),
+                ..
+            } = &mut plan.source.data_source
+            {
+                if let Err(reason) = validate.reduce_expression().await {
+                    self.metrics
+                        .webhook_validation_reduce_failures
+                        .with_label_values(&[reason])
+                        .inc();
+                }
+            }
```

Comment:
This seems like something we should be doing in planning. Any particular reason to put it in sequencing instead?

Reply:
It might be actually good that this happens in sequencing. I think @aalexandrov's long-term plan is to refactor all similar calls (that create or manipulate `MirScalarExpr`s and/or `MirRelationExpr`s) to happen in sequencing, and make "planning" go strictly until HIR.